packages/keyring-eth-ledger-bridge/src/ledger-keyring.ts

```diff
@@ -211,9 +209,11 @@ export class LedgerKeyring extends EventEmitter {
     this.hdPath = hdPath;
   }
 
-  async unlock(hdPath?: string, updateHdk = true): Promise<string> {
+  async unlock(hdPath?: string, updateHdk = true): Promise<Hex> {
     if (this.isUnlocked() && !hdPath) {
-      return 'already unlocked';
+      return this.#getChecksumHexAddress(
+        ethUtil.publicToAddress(this.hdk.publicKey, true).toString('hex'),
+      );
```

Comment:
I'm not sure how to handle this case here. The original code returned `already unlocked` if:
- there is already a `this.hdk` defined
- `hdPath` is not passed to the function (i.e. the consumer calls `keyring.unlock()`

However, since we want this function to return a value of type `Hex` we must return a valid address. Based on the lines below the actual `publicKey` returned from this function comes from the device (`payload.publicKey`) and it's used to construct an HDKey to be stored in `this.hdk`. So, the new code assumes that we want to return the address derived from the public key stored in `this.hdk.publicKey`

Reply:
Your new logic seems fine (and way better) to me. IDK if we were using the `already unlocked` value somewhere else, but that logic was kinda "fragile" IMO.

Reply:
I would just add a comment explaining this maybe? Like we re-use the already derived public key in that case (not sure how to phrase that though..)

Reply:
Definitely! done in [aa850a3](https://github.com/MetaMask/accounts/pull/194/commits/aa850a3a7eb58790f6b0f7eb1f19328d6f1a7979)